coderd/userauth.go

```diff
@@ -390,6 +391,35 @@ func (api *API) oauthLogin(r *http.Request, params oauthLoginParams) (*http.Cook
 				organizationID = organizations[0].ID
 			}
 
+			_, err := tx.GetUserByEmailOrUsername(ctx, database.GetUserByEmailOrUsernameParams{
+				Username: params.Username,
+			})
+			if err == nil {
+				var (
+					original      = params.Username
+					validUsername bool
+				)
+				for i := 0; i < 10; i++ {
+					params.Username = fmt.Sprintf("%s_%s", original, namesgenerator.GetRandomName(10))
```

Comment:
I believe this could result in a really long name, right? Because we aren't checking it against validation again.

Reply:
Good point